app/admin/top-issues/components/TopIssueDisplay.js

```diff
@@ -0,0 +1,101 @@
+'use client'
+
+import React, { useState } from 'react';
+import { SVGIconChooser } from '@shared/buttons/SVGIconChooser';
+import TextField from '@shared/inputs/TextField';
+import { ButtonGroup } from '@mui/material';
+import Button from '@mui/material/Button';
+import { FaCheck, FaEdit } from 'react-icons/fa';
+import { FaXmark } from 'react-icons/fa6';
+import Image from 'next/image';
+import IconButton from '@mui/material/IconButton';
+import { updateTopIssue } from './TopIssuesList';
+import { useTopIssues } from './UseTopIssuesContext';
+
+const insertItemInArray = (arr, item, placement) => [
+  ...arr.slice(0, placement),
+  item,
+  ...arr.slice(placement + 1)
+]
+
+export const TopIssueDisplay = ({
+  issue
+}) => {
+  const [topIssues, setTopIssues] = useTopIssues()
+  const [editTopIssueId, setEditTopIssueId] = useState(null);
+  const [editTopIssueName, setEditTopIssueName] = useState(null);
+  const [editTopIssueIcon, setEditTopIssueIcon] = useState(null);
+
+  const handleIssueEdit = issue => () => {
+    setEditTopIssueId(issue.id);
+    setEditTopIssueName(issue.name);
+    setEditTopIssueIcon(issue.icon);
+  };
+
+  const handleClearIssueEdit = () => {
+    setEditTopIssueId(null);
+    setEditTopIssueName(null);
+    setEditTopIssueIcon(null);
+  };
+
+  const handleTopIssueUpdate = (issue) => async () => {
+    const updateIssue = {
+      ...issue,
+      id: editTopIssueId,
+      name: editTopIssueName,
+      icon: editTopIssueIcon,
+    };
+    await updateTopIssue(updateIssue);
+    const issueIndx = topIssues.findIndex(({ id }) => id === issue.id);
+    setTopIssues(insertItemInArray(
+      topIssues,
+      updateIssue,
+      issueIndx
+    ));
+    handleClearIssueEdit();
+  };
+
+  return editTopIssueId === issue.id ?
+    <>
+      <SVGIconChooser
+        svgData={editTopIssueIcon}
+        setSvgData={data => setEditTopIssueIcon(data)}
+      />
+      <TextField
+        className='mx-2'
+        fullWidth={true}
+        primary={true}
+        size='small'
+        label='Top Issue Name'
+        value={editTopIssueName}
+        onChange={(e) => setEditTopIssueName(e.target.value)}
+      />
+      <ButtonGroup
+        className='mr-2'
+        variant='outlined'
+      >
+        <Button onClick={handleTopIssueUpdate(issue)}>
```

Comment:
oh I see why you did it like that. this calls the function right away.
Our conversion for this is this
```
<Button onClick={(e)=>{handleTopIssueUpdate(issue)}}>

```
This applies for all the others too.

Reply:
No, it's not calling the handler function right away. 

It's creating a thunk (https://daveceddia.com/what-is-a-thunk/) to allow for the the context of the data to be operated on (in this case, the iterated `issue`) to be passed along to the handler function, via closure scope, that's called in the future when the event handler is dispatched.

In the end, it's a way to keep from having anonymous functions throughout our JSX, and can also help w/ stack traces, since the thunks are named methods that a stack w/ spit out.